account_factoring_receivable_balance_factofrance/models/account_journal.py

```diff
@@ -0,0 +1,15 @@
+# © 2024 Open Source Integrators, Daniel Reis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import fields, models
+
+
+class AccountJournal(models.Model):
+    _inherit = "account.journal"
+
+    factor_type = fields.Selection(
+        selection_add=[("factofrance", "FactoFrance")],
+        ondelete={"factofrance": "set null"},
+    )
+    partner_identification_type = fields.Selection(
+        selection=[("1", "1 SIRET"), ("4", "4 TVA")],
+    )
```

Comment:
@dreispt  FYI in my repo I have made this settings to match with arbitrary factor keys.
![image](https://github.com/user-attachments/assets/adfb2918-e57f-4622-9aa7-a65552ab1255)

https://github.com/akretion/odoo-factoring/pull/21/files#diff-2bf0078cef96b202cab38f4dcae616055927c9c3e252cc6f6b1f74a280737c02R18-R39

May be it could be generic

Reply:
IMO that is a bit complicated.
I would go for the simpler solution, and add those additional settings as fields, only visible for tis Factor Type.
Note that the "Factor Code" is really a "Factor Contract Number", so renaming it could make sense.

Reply:
It depends on your factor settings, in your case you only have one key but like any carrier dev you may have misc keys with unpredictable name or mean unshareable. If you have several factors for different companies in your ERP it becomes crazy.
But yes,  it could be not so generic.

Reply:
Well, the DB will have all those fields, but the user form would show only the relevant ones for the selected Factor.
Like what the shipping connectors do, as you pointed out.

Reply:
A matter of taste